components/webhook-mgt/org.wso2.carbon.identity.webhook.management/src/main/java/org/wso2/carbon/identity/webhook/management/internal/dao/WebhookManagementDAO.java

```diff
@@ -97,18 +99,40 @@ public interface WebhookManagementDAO {
     /**
      * Enable a webhook subscription in the database.
      *
-     * @param webhookId Webhook subscription ID.
-     * @param tenantId  Tenant ID.
-     * @throws WebhookMgtException If an error occurs while enabling the webhook.
+     * @param webhookId       Webhook subscription ID.
+     * @param tenantId        Tenant ID.
+     * @param channels        List of channels to retry.
+     * @param status          Status of the webhook (e.g., PENDING_ACTIVATION, PENDING_DEACTIVATION).
+     * @param webhookEndpoint Webhook endpoint URL.
+     * @throws WebhookMgtException If an error occurs while retrying the webhook.
      */
-    public void activateWebhook(String webhookId, int tenantId) throws WebhookMgtException;
+    public void activateWebhook(String webhookId, int tenantId, List<Subscription> channels, WebhookStatus status,
+                                String webhookEndpoint) throws WebhookMgtException;
 
     /**
      * Disable a webhook subscription in the database.
      *
-     * @param webhookId Webhook subscription ID.
-     * @param tenantId  Tenant ID.
-     * @throws WebhookMgtException If an error occurs while disabling the webhook.
+     * @param webhookId       Webhook subscription ID.
+     * @param tenantId        Tenant ID.
+     * @param channels        List of channels to retry.
+     * @param status          Status of the webhook (e.g., PENDING_ACTIVATION, PENDING_DEACTIVATION).
+     * @param webhookEndpoint Webhook endpoint URL.
+     * @throws WebhookMgtException If an error occurs while retrying the webhook.
+     */
+    public void deactivateWebhook(String webhookId, int tenantId, List<Subscription> channels, WebhookStatus status,
+                                  String webhookEndpoint) throws WebhookMgtException;
+
+    /**
+     * Retry a webhook subscription or unsubscription that has failed.
+     *
+     * @param webhookId       Webhook subscription ID.
+     * @param tenantId        Tenant ID.
+     * @param channels        List of channels to retry.
+     * @param status          Status of the webhook (e.g., PENDING_ACTIVATION, PENDING_DEACTIVATION).
+     * @param webhookEndpoint Webhook endpoint URL.
+     * @throws WebhookMgtException If an error occurs while retrying the webhook.
      */
-    public void deactivateWebhook(String webhookId, int tenantId) throws WebhookMgtException;
+    public void retryWebhook(String webhookId, int tenantId, List<Subscription> channels, WebhookStatus status,
```

Comment:
Move tenantId to the end

Reply:
fixed with https://github.com/wso2/carbon-identity-framework/pull/6885/commits/fb575c056228c93d2221b4ea29c500a76d09311c